pages/[resourceType]/index.tsx

```diff
@@ -1,85 +1,147 @@
 import { useEffect, useState } from "react";
 import { useRouter } from "next/router";
 import ResourceIDs from "../../components/ResourceIDs";
-import { Center, Loader, Stack } from "@mantine/core";
+import { Center, Divider, Loader, Pagination } from "@mantine/core";
 import { fhirJson } from "@fhir-typescript/r4-core";
 import { Button } from "@mantine/core";
 import Link from "next/link";
+import { textGray } from "../../styles/appColors";
+
+const NUMBER_IDS_RENDERED_AT_A_TIME = 10;
 
 /**
  * Component page that renders Buttons for all IDs of a resourceType. A request is made to
  * the test server to retrieve all resources of a specified type. Then a ResourceID component is
  * returned with the Buttons for each resourceID or a "No resources found" message is displayed, or if
  * the request to the server throws an error then a "Problem connecting to server" message is displayed.
- * Loader is displayed while http request is in progress
- * @returns Loader, ResourceIDs component, or error message depending on the status of the http request made.
+ * Loader is displayed while http request is in progress. For resourceTypes with more than 10 resources,
+ * Pagination component is implemented to allow for access to all resources.
+ * @returns Loader, ResourceIDs and Pagination components, or error message depending on the status of the http request made.
  */
 function ResourceTypeIDs() {
-  //get resourceType from the current url with useRouter
   const router = useRouter();
-  const { resourceType } = router.query;
-
+  const { resourceType, page } = router.query;
   const [pageBody, setPageBody] = useState<fhirJson.Bundle>();
   const [fetchingError, setFetchingError] = useState(false);
   const [loadingRequest, setLoadingRequest] = useState(false);
+
+  const [activePageNum, setActivePageNum] = useState<number>(1);
+
   useEffect(() => {
-    if (resourceType) {
-      setLoadingRequest(true);
-      fetch(`${process.env.NEXT_PUBLIC_DEQM_SERVER}/${resourceType}`)
-        .then((data) => {
-          return data.json() as Promise<fhirJson.Bundle>;
-        })
-        .then((resourcePageBody) => {
-          setPageBody(resourcePageBody);
-          setFetchingError(false);
-          setLoadingRequest(false);
-        })
-        .catch((error) => {
-          console.log(error.message);
-          setFetchingError(true);
-          setLoadingRequest(false);
-        });
+    let pageUrl = `${resourceType}`;
+    if (page != null) {
```

Comment:
```suggestion
    if (page) {
```
I think? More concise.